src/extension/runner.ts

```diff
@@ -98,23 +106,46 @@ export interface IRunnerProgramSession extends DisposableAsync, Pseudoterminal {
 }
 
 export class GrpcRunner implements IRunner {
-  protected readonly client: RunnerServiceClient
-  protected transport: GrpcTransport
+  protected client?: RunnerServiceClient
 
-  private children: WeakRef<DisposableAsync>[] = []
+  private children: WeakRef<IRunnerChild>[] = []
+  private disposables: Disposable[] = []
 
-  constructor() {
-    this.transport = new GrpcTransport({
-      host: getGrpcHost(),
-      channelCredentials: ChannelCredentials.createInsecure(),
-    })
+  protected _onReady = this.register(new EventEmitter<void>())
+  onReady = this._onReady.event
 
-    this.client = new RunnerServiceClient(this.transport)
+  constructor(protected server: RunmeServer) {
+    this.disposables.push(
+      server.onTransportReady(({ transport }) => this.initRunnerClient(transport))
+    )
+
+    this.disposables.push(
+      server.onClose(() => this.deinitRunnerClient())
+    )
+  }
+
+  private deinitRunnerClient() {
+    this.disposeChildren()
+    this.client = undefined
+  }
+
+  private initRunnerClient(transport?: GrpcTransport) {
+    this.deinitRunnerClient()
+    this.client = new RunnerServiceClient(transport ?? this.server.transport())
+    this._onReady.fire()
+  }
+
+  protected static assertClient(client: RunnerServiceClient|undefined): asserts client {
+    if(!client) {
+      throw new Error('Client is not active!')
```

Comment:
Could this be something a user is exposed to? In that case showing an `window.showErrorMessage` might be more appropiate.

Reply:
Fair point. I assumed it's caught "top of stack," but that might not be the case always, @mxsdev. However, since it's all in the Runner (which is default deactivated), let's do this in a follow-up issue. Let's get this merged. We need to get a release out today.